clerk.go

```diff
@@ -219,7 +224,11 @@ func (b *defaultBackend) newRequest(ctx context.Context, apiReq *APIRequest) (*h
 	}
 	req.Header.Add("Authorization", fmt.Sprintf("Bearer %s", secretKey))
 	req.Header.Add("Content-Type", "application/json")
-	req.Header.Add("User-Agent", fmt.Sprintf("Clerk/%s SDK-Go/%s", clerkAPIVersion, sdkVersion))
+	userAgent := fmt.Sprintf("Clerk/%s SDK-Go/%s", clerkAPIVersion, sdkVersion)
```

Comment:
ℹ️ Just mentioning that Cloudflare analytics group by user agent, but I guess we can always filter by prefix!

Reply:
Thanks, I didn't know that. Do you think it's better to drop this change. 

An alternative solution would be to leave the User-Agent string as is and maybe send a custom header instead. Would that be better?

Reply:
@georgepsarakis Can't we just use the `X-Clerk-SDK` header?

Reply:
The `X-Clerk-SDK` is already used, but we could add another one for debugging purposes, or to uniquely identify a consumer, like one of our partners or us.

Reply:
Yep, what I had in mind was to let the user set the User-Agent to whatever, not just the suffix, and rely on the `X-Clerk-SDK` for wherever we need to identify if it came from the SDK.

Reply:
I see. That could work. However, I think we should have control over the User-Agent header, since it's a "standard" header but be loose in what we accept for custom ones.

Reply:
@agis we are following this approach in for BAPI requests by setting the request user-agent to `{packageName}@{packageVersion}`.

ref: https://github.com/clerk/javascript/blob/main/packages/backend/src/constants.ts#L4

Reply:
Here's an alternative, which uses a custom header. https://github.com/clerk/clerk-sdk-go/pull/252